girder/molecules/server/utilities/molecules.py

```diff
@@ -59,6 +59,9 @@ def create_molecule(data_str, input_format, user, public):
 
         smiles = openbabel.to_smiles(sdf_data, sdf_format)
 
+        # Generate an svg file for an image
+        svg_data = openbabel.to_svg(sdf_data, sdf_format)
```

Comment:
I'm not sure how long it takes to generate the svg from the 3d coordinates, but if it takes too long, it could be a good idea to generate the svg in an async job. We do something similar here:
https://github.com/OpenChemistry/edp/blob/master/server/edp/server/models/base.py#L135
If you look at that example, we are checking if there are any images attached to a specific document, and if there are we spawn jobs to generate the thumbnail for each. Once the jobs are complete, they modify the document to add a reference to the new thumbnail file id.

Reply:
I haven't tried it on huge systems, but a molecule with 60 atoms for me is still nearly instant on my computer. We can keep an eye on whether or not the speed becomes an issue.

Reply:
Shouldn't be too bad, we can keep it simple for now. The 3d coordinate generation would do with async, much more expensive call!